python_ta/transforms/type_inference_visitor.py

```diff
@@ -196,22 +196,18 @@ def _closest_frame(self, node, name):
     # Name lookup and assignment
     ##############################################################################
     def visit_name(self, node: astroid.Name) -> None:
-        try:
-            node.inf_type = self.lookup_inf_type(node, node.name)
-        except KeyError:
+        node.inf_type = self.lookup_inf_type(node, node.name)
```

Comment:
Store this in a temporary variable. Prefer symmetry in branches for initialization (i.e., add an extra branch below for `node.inf_type = ___` if the value is _not_ a `TypeFail`.